src/workflows/airqo_etl_utils/datautils.py

```
@@ -52,18 +52,25 @@ def extract_devices_data(
         local_file_path = "/tmp/devices.csv"
 
         # Load devices from cache
-        devices = DataUtils.load_cached_data(local_file_path, MetaDataType.DEVICES.str)
-        devices["device_number"] = devices["device_number"].fillna(-1)
+        try:
+            (
+                devices := DataUtils.load_cached_data(
+                    local_file_path, MetaDataType.DEVICES.str
+                )
+            )["device_number"] = devices["device_number"].fillna(-1)
```

Ruff (0.8.2)

Undefined name `devices`

(F821)